src/Relude/Extra/Foldable1.hs

```diff
@@ -1,54 +1,116 @@
+{-# LANGUAGE InstanceSigs #-}
+
 module Relude.Extra.Foldable1
     ( Foldable1 (..)
-    , traverse1_
-    , for1_
     ) where
 
-import Relude.Applicative (Applicative, pass)
-import Relude.Base (Ord)
-import Relude.Foldable.Reexport (Foldable (foldr))
-import Relude.Function (flip, id, (.))
-import Relude.Functor.Reexport (Compose (..), Identity (..))
-import Relude.Monoid (Semigroup, (<>))
+import Relude hiding (Product (..), Sum (..))
+
+import Data.Functor.Product (Product (..))
+import Data.Functor.Sum (Sum (..))
+import qualified Data.Semigroup as SG
 
-import Control.Applicative ((*>))
-import Data.List.NonEmpty (NonEmpty (..))
-import Data.Semigroup (First (..), Last (..), Max (..), Min (..))
+-- | The class of foldable data structures that cannot be empty.
 
 class Foldable t => Foldable1 t where
-    foldMap1   :: Semigroup m => (a -> m) -> t a -> m
+    {- | Map each element of the non-empty structure to a semigroup, and combine the results.
+
+    >>> foldMap1 SG.Sum (1 :| [2, 3, 4])
+    Sum {getSum = 10}
+    -}
+    foldMap1 :: Semigroup m => (a -> m) -> t a -> m
 
+    {- | Combine the elements of a non-empty structure using a semigroup.
+
+    >>> fold1 (4 :| [5, 10 :: SG.Product Int])
```

Comment:
I think it's better to write first example for this function like:

```haskell
fold1 @SG.Sum (1 :| [2, 3, 4])
```

And only then have your example with `Product`. It's a nice parallel with `foldMap1` example.